pkg/process/metadata/workloadmeta/grpc.go

```diff
@@ -113,6 +115,15 @@ func (l *GRPCServer) StreamEntities(_ *pbgo.ProcessStreamEntitiesRequest, out pb
 				continue
 			}
 
+			// The diff received from the channel should be 1 + the previous version. Otherwise, we have lost data,
+			// and we should signal the client to resync by closing the stream.
+			log.Trace("[WorkloadMeta GRPCServer] expected diff version %v, actual %v", expectedVersion, diff.cacheVersion)
+			if diff.cacheVersion != expectedVersion {
+				log.Debug("[WorkloadMeta GRPCServer] missing cache diff - dropping stream")
+				return fmt.Errorf("missing cache diff: received version = %v; expected = %v", diff.cacheVersion, expectedVersion)
```

Comment:
Curious if we should capture this in a status field so that the user can see these errors in the flare?

Reply:
That is definitely possible, however, given that the feature hasn't GA'ed and we will add prometheus telemetry for 7.47 for now I'd like to keep this as a follow up given that the freeze is in 1 day and we still have 1 more PR that we have to review and merge.